### DISCUSSION TITLE
How to limit the file size uploaded by the user through Theia, for example, the user selects a file larger than 100M prompting ‘not allowed’.

### DISCUSSION THREAD

**Discussion by MrZmcH** [eclipse-theia/theia#12718] [Q&A]
I wonder if anyone can help me.
![image](https://github.com/eclipse-theia/theia/assets/45864434/bd691fa4-ff70-44db-a9d5-1003844c995c)

**Comment by vince-fugnitto** [comment]
@MrZmcH thanks for the discussion, you'll likely want to extend `NodeFileUploadService` in a [custom extension](https://theia-ide.org/docs/authoring_extensions/) so it takes into consideration the file size before actually performing the upload:

https://github.com/eclipse-theia/theia/blob/3deecfe17c54916886b06f2fb80a47dc619d5490/packages/filesystem/src/node/node-file-upload-service.ts#L27

https://github.com/eclipse-theia/theia/blob/3deecfe17c54916886b06f2fb80a47dc619d5490/packages/filesystem/src/node/node-file-upload-service.ts#L59-L78

The size should be available from `request.file.size` in this case.

You can find this related implementation by searching for `upload files...` and understanding how it is implemented, and seeing at which point you can/should add a limit.

---

Note that I deleted the other duplicate discussions opened. In general the community is happy to help if they have knowledge for the question and/or time to help so no need to open duplicate questions.
